tests-ng/plugins/initrd.py

```diff
@@ -0,0 +1,246 @@
+import logging
+import os
+import re
+import tempfile
+from typing import List, Optional
+
+import pefile
+import pytest
+
+from .kernel_versions import KernelVersions
+from .shell import ShellRunner
+from .utils import get_cname_from_os_release
+
+logger = logging.getLogger(__name__)
+
+
+class Initrd:
+    """Inspect initrd contents using lsinitrd (dracut)."""
+
+    def __init__(self, shell: ShellRunner, kernel_versions: KernelVersions):
+        self._shell = shell
+        installed = kernel_versions.get_installed()
+        if not installed:
+            self._kernel_version = None
+            self._contents: List[str] = []
+            return
+
+        self._kernel_version = installed[0].version
+        self._contents = self._load_initrd_contents(self._kernel_version)
+
+    def _extract_initrd_from_efi(self) -> Optional[str]:
+        """Extract initrd from EFI file for trustedboot/USI flavors.
+
+        The initrd is embedded in the EFI file as a PE32+ section named '.initrd'.
+
+        Returns:
+            Path to extracted initrd file, or None if extraction fails.
+        """
+        cname = get_cname_from_os_release()
+        if not cname:
+            logger.debug("CNAME not found in /etc/os-release, cannot locate EFI file")
+            return None
+
+        efi_path = f"/efi/EFI/Linux/{cname}.efi"
+        if not os.path.exists(efi_path):
+            logger.debug(f"EFI file not found: {efi_path}")
+            return None
+
+        try:
+            logger.debug(f"Loading EFI file: {efi_path}")
+            pe = pefile.PE(efi_path, fast_load=True)
+
+            # Find the .initrd section
+            initrd_section = None
+            for section in pe.sections:
+                # Section names are null-padded 8-byte strings
+                section_name = section.Name.rstrip(b"\x00").decode(
+                    "ascii", errors="ignore"
+                )
+                if section_name == ".initrd":
+                    initrd_section = section
+                    logger.debug(f"Found .initrd section in EFI file: {efi_path}")
+                    break
+
+            if not initrd_section:
+                logger.warning(f"No .initrd section found in EFI file: {efi_path}")
+                return None
+
+            # Extract section data
+            initrd_data = initrd_section.get_data()
+            if not initrd_data or len(initrd_data) == 0:
+                logger.warning(f".initrd section is empty in EFI file: {efi_path}")
+                return None
+
+            # Write extracted data to a temporary file
+            with tempfile.NamedTemporaryFile(delete=False, suffix=".img") as tmp_file:
```

Comment:
As you don't use advanced features of the `NamedTemporaryFile()` class you may just use `mkstemp()` as well.